### PR TITLE
Revised initial conditions for convergence test simulation

### PR DESCRIPTION
**Pull request by zhangshixuan1987**
Hi Vince, 

This commit contains code changes related to the modified initial conditions for convergence test simulations. These code changes can be activated by setting **l_modify_ic_for_cnvg_test = .true.**  in the CLUBB namelist.  Currently, these changes have been tested for BOMEX, RICO, DYCOMS2_RF02 and Wangara cases. 
    
    The modified files for this implementation include:
      ../src/CLUBB_core/clubb_api_module.F90
      ../src/CLUBB_core/model_flags.F90
      ../src/G_unit_test_types/pdf_parameter_tests.F90
      ../src/G_unit_test_types/spurious_source_test.F90
      ../src/clubb_driver.F90
      ../src/clubb_tuner.F90
      ../src/sounding.F90

### CODE REVIEW

File: src/sounding.F90 [hunk 1]
```diff
@@ -320,29 +327,107 @@ subroutine read_sounding( gr, iunit, runtype, p_sfc, zm_init,&
...
-                                    sclr(k,j), sclr(k-1,j) )
-            end do
-          end if
-          if ( edsclr_dim > 0 ) then
-            do j = 1, edsclr_dim
-              edsclrm(i,j) = lin_interpolate_two_points( gr%zt(1,i), z(k), z(k-1),  & 
-                                      edsclr(k,j), edsclr(k-1,j) )
-            end do
-          end if
+        IF ( trim( runtype ) /= "dycoms2_rf02" .or. l_modify_ic_for_cnvg_test ) THEN
```
Comment (vlarson) [inline src/sounding.F90:330]: Do we only need to smooth the sounding for the dycoms2_rf02 case?  Do the wangara, rico, and bomex cases converge without smoothing the initial condition?

Comment (zhangshixuan1987) [reply src/sounding.F90:330]: @vlarson: for our convergence test simulation in the paper, we need to smooth the initial condition profiles for all the cases:
1. For RICO, BOMEX and Wangara case, we employ cubic spline interpolation to remap the sounding profiles to the CLUBB's grid to obtain smoother initial condition profiles 
2. For Dycoms_RF02 case: the default setup in CLUBB uses the formulations built in the src/sounding.F90 to derive the initial conditions, which are space-dependent. In our investigation, I choose to construct the initial condition profile using the same formulations in the code, and saved them in  **dycoms2_rf02_sounding.in** file. Then all simulations read the **dycoms2_rf02_sounding.in**  file for initialization  
3. there are also some changes for *_soundign.in files, I choose to do the changes in the simulation run scripts.  I will submit a separate pull request for the scripts to run convergence test simulations

**Comment by vlarson** [comment]
This looks good to me.

-  Does running the dycoms2_rf02 case at standard resolution with `l_modify_ic_for_cnvg_test=T` and `l_modify_ic_for_cnvg_test=F` produce similar-looking results?  

-  Does running, e.g., the bomex case at standard resolution produce bit-for-bit identical results with 
`l_modify_ic_for_cnvg_test=T` and `l_modify_ic_for_cnvg_test=F`?

-  UWM will need to think about how best to merge this pull request from the clubb-release repo back into the clubb repo.  @huebleruwm, any thoughts?

-  Normally, we include all important flags in `clubb_config_flags`.  However, I question if including the flag `l_modify_ic_for_cnvg_test` in `clubb_config_flags` is worthwhile.   `l_modify_ic_for_cnvg_test`  is more of an initialization flag than a flag that affects the timestep-by-timestep evolution of CLUBB.  The E3SM folks will wonder how to set it, when in fact the flag has no impact on global simulations.  Maybe the flag should just be hard-wired in sounding.F90.  @bmg929, thoughts?

-  However, I hasten to add that any flag that will affect the results of a global E3SM run should be included in clubb_config_flags.  `l_modify_ic_for_cnvg_test` is an exception.

**Comment by zhangshixuan1987** [comment]
@vlarson: please give me some time, I will provide some figures to answer your question above tomorrow.  For the implementation on UMW side, please feel free to make any changes, and let me know how should I coordinate with you.

**Comment by zhangshixuan1987** [comment]
I closed this pull request and submit a new one which contains the revisions following Vince's comments.

**Comment by zhangshixuan1987** [comment]
@vlarson please check the pull request https://github.com/larson-group/clubb_release/pull/9 for the change following your comments. Let us finish this commit first, then I will resubmit the pull request for subsequent changes (e.g. boundary conditions etc.)

**Pull request closed without merging**